tsconfig.json

```diff
@@ -30,7 +30,7 @@
     "strictNullChecks": true,
     "strictPropertyInitialization": false,
     "suppressImplicitAnyIndexErrors": true,
-    "target": "es5"
+    "target": "es2017"
```

Comment:
https://stackoverflow.com/a/61037142

Reply:
I installed `es-check` to check whether or not our existing builds are compatible with es5. 

When running against both `iam-client-lib.esm.js` and `iam-client-lib.js`, both reported that they aren't currently compatible with es5.

```
❯ es-check es5 dist/iam-client-lib.js          
ES-Check: there were 1 ES version matching errors.

          ES-Check Error:
          ----
          · erroring file: dist/iam-client-lib.js
          · error: SyntaxError: The keyword 'let' is reserved (1070:0)
```

```
❯ es-check es5 dist/iam-client-lib.esm.js
ES-Check: there were 1 ES version matching errors.

          ES-Check Error:
          ----
          · erroring file: dist/iam-client-lib.esm.js
          · error: SyntaxError: The keyword 'import' is reserved (1:0)
```

So maybe our existing library isn't compatible with es5 🤔  ?

Reply:
For iam-client-lib to be compatible with es5, would all of the node_modules used by iam-client-lib also need to be es5 compatible?

Reply:
after doing some more checking, I think we're current only compatible with es9 actually. I think this is due to the bundling of the nats client: 
![image](https://user-images.githubusercontent.com/4407464/110967617-d40a6380-8313-11eb-9520-fb677bbf6528.png)